libsolidity/analysis/DeclarationTypeChecker.cpp

```diff
@@ -291,6 +292,16 @@ void DeclarationTypeChecker::endVisit(ArrayTypeName const& _typeName)
 
 	if (Expression const* length = _typeName.length())
 	{
+		if (auto literal = dynamic_cast<Literal const*>(length))
+		{
+			if (!holds_alternative<Literal::SubDenomination>(literal->suffix()))
+				m_errorReporter.typeError(
+					4127_error,
+					literal->location(),
+					"Array length cannot contain a suffix function."
+				);
```

Comment:
Can you merge this with error 5462 (`Invalid array length, expected integer literal or constant expression.`)? I think that both situations should report the same error ID. No point defining a special one just for suffixes.

If you really want, you could add something like this if you detect that it has a suffix: `A suffixed literal is not a constant expression unless the suffix is a denomination`. But it would be completely fine even without it.

Reply:
Merged and appended a message addressing the suffixed literal error case.